### PR TITLE
Add support for SSL

### PR DESCRIPTION
**Pull request by legiangthanh**
The current implementation doesn't support SSL(TLS). This PR will add two options to support TLS

tls_enable: enable/disable tls (boolean, optional, default false)
tls_insecure: if false, certificate and host is validated. Otherwise, skip validated (boolean, optional, default false)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,8 @@ This plugin only works with embulk >= 0.8.8.
     - **user**: (string, optional)
     - **password**:  (string, optional)
     - **database**:  (string, required)
+    - **tls_enable**: enable/disable tls (boolean, optional, default false)
+    - **tls_insecure**: enable/disable certificate and host validation (boolean, optional, default false)
```
Comment (dmikurube) [inline README.md:30]: @legiangthanh Have you confirmed similar option names of other plugins (especially OSS plugins under https://github.com/embulk/)?  We should have consistent names with other plugins as far as possible.

* I know there already are some variations, but I have never seen `tls_enable`.
* `ftp` and JDBC plugins use just `ssl`.
* "SSL" is deprecated in general, so `tls` could be reasonable.
* `ftp` has `ssl_verify`, which seems similar (but opposite) to `tls_insecure`?

Comment (legiangthanh) [reply README.md:30]: @dmikurube , 
```
I know there already are some variations, but I have never seen tls_enable.
ftp and JDBC plugins use just ssl.
"SSL" is deprecated in general, so tls could be reasonable.
```
Sorry, I don't notice option names of other plugins, I will change it to ssl.

`ftp has ssl_verify, which seems similar (but opposite) to tls_insecure?`
This option is similar option supported by mongo command, please refer https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tlsInsecure

Comment (dmikurube) [reply README.md:30]: As I mentioned, `ssl` -> `tls` could be reasonable in this age. But `_enable` has not been common for other plugins.

> This option is similar option supported by mongo command, please refer
> https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tlsInsecure

Got it... that is a point of design choice. Then `tls_insecure` sounds reasonable, but nice to explain it ("the same with `mongo` command") both in `README` and `PluginType`.

File: README.md [hunk 2]
```diff
@@ -26,6 +26,8 @@ This plugin only works with embulk >= 0.8.8.
     - **user**: (string, optional)
     - **password**:  (string, optional)
     - **database**:  (string, required)
+    - **tls**: enable/disable tls (boolean, optional, default false)
+    - **tls_insecure**: enable/disable certificate and host validation (boolean, optional, default false), this option similar to option supported by mongo command (https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tlsInsecure)
```
Comment (dmikurube) [inline README.md:30]: Some updates on the English statement.

```suggestion
    - **tls**: `true` to use TLS to connect to the host (boolean, optional, default: `false`)
    - **tls_insecure**: `true` to disable various certificate validations (boolean, optional, default: `false`)
      - The option is similar to an option of the official `mongo` command.
      - See also: https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tlsInsecure
```

When users see "enable/disable tls", they are not sure `true` to enable or `true` to disable. 

Even if the think `true` for the former (`enable`) for this, "enable/disable certificate and host validation" is the opposite. It's `true` for the latter (`disable`).

It's so much confusing for users. A general documenting technique for `boolean` values is to explain like :
- "`true` if ..." or
- "`true` to ...".

File: src/main/java/org/embulk/input/mongodb/MongodbInputPlugin.java [hunk 3]
```diff
@@ -324,10 +330,52 @@ private MongoClient createClientFromParams(PluginTask task)
...
+            builder.sslEnabled(true);
+            if (task.getTlsInsecure()) {
+                builder.sslInvalidHostNameAllowed(true);
+                builder.sslContext(createAcceptAllCertificate());
+            }
+        }
+        return builder.build();
+    }
+
+    private SSLContext createAcceptAllCertificate()
```
Comment (dmikurube) [inline src/main/java/org/embulk/input/mongodb/MongodbInputPlugin.java:353]: nitpick: The method name is difficult to understand: verb (create) + verb (accept).

How about: `createSSLContextToAcceptAnyCert` ?

File: src/main/java/org/embulk/input/mongodb/PluginTask.java [hunk 4]
```diff
@@ -39,6 +39,15 @@
     @ConfigDefault("null")
     Optional<List<HostTask>> getHosts();
 
+    @Config("tls")
+    @ConfigDefault("false")
+    boolean getTls();
+
+    // similar to option supported by mongo command (https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tlsInsecure)
```
Comment (dmikurube) [inline src/main/java/org/embulk/input/mongodb/PluginTask.java:46]:
```suggestion
    // The option is similar to an option of the official `mongo` command.
    // (https://www.mongodb.com/docs/manual/reference/connection-string/#mongodb-urioption-urioption.tlsInsecuresimilar)
```

**Review by dmikurube — APPROVED** [review]
Thanks for working on it. Approved to unblock, but left another couple of comments (for English statements).

**Comment by dmikurube** [comment]
I'll merge this and release v0.8.2 once I confirm @hieudion's review.

**Review by hieudion — APPROVED** [review]
LGTM

**Comment by dmikurube** [comment]
Merging, and releasing v0.8.2 soon...

**Pull request merged**